Userland/Libraries/LibJS/Runtime/Value.h

```diff
@@ -150,18 +150,8 @@ class Value {
         }
     }
 
-    explicit Value(unsigned long value)
-    {
-        if (value > NumericLimits<i32>::max()) {
-            m_value.as_double = static_cast<double>(value);
-            m_type = Type::Double;
-        } else {
-            m_value.as_i32 = static_cast<i32>(value);
-            m_type = Type::Int32;
-        }
-    }
-
-    explicit Value(unsigned value)
+    template<typename T>
+    requires(IsIntegral<T> && !IsSameIgnoringCV<T, i32> && !IsSameIgnoringCV<T, bool>) explicit Value(T value)
     {
         if (value > NumericLimits<i32>::max()) {
             m_value.as_double = static_cast<double>(value);
```

Comment:
This is incorrect, since we're only checking `IsIntegral` signed values are now also accepted, and for those that are larger than `i64`, this check will cause unintended overflows (for example `i64`)

Reply:
Good catch, thanks! excluding signed i32/i64 should be enough, right?

Reply:
It should, yes. The check is technically still incorrect, since we're only checking that the value is not larger than the maximum of i32, we don't check it's not smaller than the minimum, but with these constraints no such type exists.